src/pages/HomePage.tsx

```diff
@@ -42,47 +39,32 @@ export const HomePage: React.FC = () => {
     }
   };
 
-  const toggleDialog = () => {
-    setIsDialogOpen(!isDialogOpen);
-  };
-
-  const resetSelectedMovie = () => {
-    setSelectedMovie(null);
-  };
-
   const handleMovieClick = (movie: MovieInfo) => {
     navigate(`movies/${movie.id}?search=${searchQuery}&genre=${selectedGenre}&sortBy=${sortBy}`);
   };
 
+  useEffect(() => {
+    if (pathname === '/') {
+      fetchMovies();
+    }
+  }, [pathname, fetchMovies]);
+
   return (
     <div>
-      {selectedMovie ? (
-        <MovieDetails movie={selectedMovie} onSearchClick={resetSelectedMovie} />
-      ) : (
+      {pathname === '/' && (
```

Comment:
pathname === '/' - you can create a separate variable for that condition and use it in the both places